addons/mail/static/src/core/common/attachment_list.js

```diff
@@ -123,6 +134,11 @@ export class AttachmentList extends Component {
 
     getActions(attachment) {
         const res = [];
+        res.push({
+            label: _t("Show in conversation"),
+            icon: "fa fa-eye",
+            onSelect: () => this.onClickShowInConversation(attachment),
+        });
```

Comment:
Feature should probably be disabled with attachments not linked to message, e.g. when uploading an attachment directly in chatter.

`message` should probably be passed to `_to_store` of attachment, so that there's always awareness of whether the feature exist when receiving attachment data from server.